scripts/language_model/model/qa.py

```diff
@@ -0,0 +1,223 @@
+"""XLNetForQA models."""
+
+import mxnet as mx
+from mxnet.gluon import HybridBlock, Block, loss, nn
+
+
+
+class PoolerStartLogits(HybridBlock):
+    """ Compute SQuAD start_logits from sequence hidden states. """
+    def __init__(self, prefix=None, params=None):
+        super(PoolerStartLogits, self).__init__(prefix=prefix, params=params)
+        self.dense = nn.Dense(1, flatten=False)
+
+    def __call__(self, hidden_states, p_masks=None):
+        # pylint: disable=arguments-differ
+        return super(PoolerStartLogits, self).__call__(hidden_states, p_masks)
+
+    def hybrid_forward(self, F, hidden_states, p_mask):
+        # pylint: disable=arguments-differ
+        """ Args:
+            **p_mask**: (`optional`) ``torch.FloatTensor`` of shape `(batch_size, seq_len)`
+                invalid position mask such as query and special symbols (PAD, SEP, CLS)
+                1.0 means token should be masked.
+        """
+        x = self.dense(hidden_states).squeeze(-1)
+        if p_mask is not None:
+            x = x * (1 - p_mask) - 1e30 * p_mask
+        return x
+
+
+class PoolerEndLogits(HybridBlock):
+    """ Compute SQuAD end_logits from sequence hidden states and start token hidden state.
+    """
+    def __init__(self, units=768, prefix=None, params=None):
+        super(PoolerEndLogits, self).__init__(prefix=prefix, params=params)
+        self.dense_0 = nn.Dense(units, activation='tanh', flatten=False)
+        self.dense_1 = nn.Dense(1, flatten=False)
+        self.layernorm = nn.LayerNorm()
+
+    def __call__(self, hidden_states, start_states=None, start_positions=None, p_masks=None):
+        # pylint: disable=arguments-differ
+        return super(PoolerEndLogits, self).__call__(hidden_states, start_states, start_positions,
+                                                     p_masks)
+
+    def hybrid_forward(self, F, hidden_states, start_states, start_positions, p_mask):
+        # pylint: disable=arguments-differ
+        assert start_states is not None or start_positions is not None
```

Comment:
Should this rather be XOR (one and only one should be not None)?

Note that due to limitation of current HybridBlock, you can't change the argument that is `None` in subsequent calls after hybridization. The computational graph will be fixed.
Does this currently work for you? My understanding is that you use `start_positions` during training and `start_states` during testing? Did you run the script with `args.only_predict is True`?